File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(number)
    if number % 3 == 0
      return "Pling"
    end

    if number % 5 == 0
      return "Plang"
    end

    if number % 7 == 0
      return "Plong"
    end

    return number.to_s
  end
end

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

const TestVersion = 1

var drops = []struct {
	number int
	sound  string
}{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

func Convert(input int) (result string) {

	for _, d := range drops {
		if input%d.number == 0 {
			result += d.sound
		}
	}

	if result == "" {
		result = fmt.Sprintf("%d", input)
	}
	return
}
